crates/wdl-analysis/src/types.rs

```diff
@@ -911,6 +949,134 @@ impl Coercible for StructType {
     }
 }
 
+/// Represents the type of an enum.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EnumType {
+    /// The name of the enum.
+    name: Arc<String>,
+    /// The common coerced type computed from all variant values.
+    value_type: Arc<Type>,
+    /// The variants with their actual types (before coercion).
+    variants: Arc<IndexMap<String, Type>>,
```

Comment:
Is there value in storing the original types of the variants vs. just a `Arc<[String]>` here?

For static analysis, evaluating a variant definition's expression (which should be a _constant expression_), the resulting type must coerce to either the declared value type of the enum or the common infered type; once that takes place, I don't see us particuarly caring about the "original" type of the variant as for all intense and purposes `value(EnumValue)` always produces a type that is the enum's value type and never what the type of the initialization expression was for a particular variant.